### ISSUE TITLE
PHP - Refactoring how Google sheets are defined

### ISSUE THREAD

**Issue by stacksavings** [StackSavingsTeam/googlesheetslayout#29]
We need to refactor how the Google sheets are defined as the current way this is setup is not scaleable. Soon we will have a production version of the sheets and a dev version so we need to have an easy way to change which ones are used by only having to change a property file.

1) create a file under /vendor/stacksavings called "google_sheet_definitions.properties" (or whatever file format extension is appropriate

This should be a key / value properties file, if possible, nothing too fancy, just like:

layout_values_sheet=<sheet_id>
facebook_submissions_sheet=<sheet_id>

NOTE: you should define the current values in this, as these will be used for testing, so we will check in sheet ID's for testing (when we deploy to production the idea is that we can just then easily overwrite this properties file with the new sheet values and that is all that is needed then to change what sheets are used in production)

2) Create a PHP function within the /vendor/stacksavings folder that can retrieve the appropriate value, as requested so that it can be retrieved in the code. 

3) so, for now, it will have these two sheets that we are using (the facebook one and the one that has the layout values). Any code that uses either of these, then, needs to be refactored to use this and not have the sheet value hard-coded but rather pull from the properties file.

**Comment by cyberdypro** [comment]
@stacksavings @avi-stacksavings i have update with commit [https://github.com/StackSavingsTeam/googlesheetslayout/commit/1a11c78066589092e54b8407504ea30f7071b150](url)

**Comment by stacksavings** [comment]
@cyberdypro Please be more clear, is this ticket done and ready for testing? Are there any outstanding things waiting on this, etc?

**Comment by cyberdypro** [comment]
@stacksavings I have add new class with name `Config` under `vendor/stacksavings/config.php` and a config file name `vendor/stacksavings/google_sheet_definitions.properties`. For now, we should be store spreadsheet id config into this file and get from any where with `Config::getInstance()->getProperty('[config_name]');`. I have updated this config into `quickstart.php`, the layout spreadsheet id should be get from `Config::getInstance()->getProperty('facebook_submissions_sheet');`.  Similarly I will update this way to facebook sheet id in ticket #29

**Comment by avi-stacksavings** [comment]
having issue while adding different sheet ID in google_sheet_definitions.properties

when trying clicking on FAcebook Login button this message pops up.
"URL blocked: This redirect failed because the redirect URI is not white-listed in the app's client OAuth settings. Make sure that the client and web OAuth logins are on and add all your app domains as valid OAuth redirect URIs."

**Comment by stacksavings** [comment]
@avi-stacksavings  need a new ticket for that issue, it has nothing to do with this ticket.

**Comment by avi-stacksavings** [comment]
Tested on both the URLs
servicejoin.com/livre-4/index.php
http://stacksavings.com/connect/livre-4/

FB button working good on both.

@stacksavings Pending final approval.

**Comment by avi-stacksavings** [comment]
layout not reading from new sheet ID opening ticket #33
